samples/corda/fungible-house-token/extension-flows/AssetTransferHelperFlows.kt

```diff
@@ -0,0 +1,251 @@
+ *
+ * @property assetPledgeState The (interop) vault state that represents the pledge details on the ledger.
+ */
+@InitiatingFlow
+@StartableByRPC
+class GetHouseTokenJsonStringFromStatePointer(
+    val assetPledgeState: AssetPledgeState
+) : FlowLogic<String>() {
+    @Suspendable
+    override fun call(): String {
+
+        if (assetPledgeState.assetStatePointer == null) {
+            // Typically, [AssetPledgeState].assetStatePointer will be null only in the case of pledge details not
+            // being available for a given pledgeId. The flow GetAssetPledgeStatus in AssetTransferFlows sets this
+            // pointer to null if the pledge-state is not available in the context of the interop-query from the
+            // importing n/w to the exporting n/w. Hence return empty string, and this will not be passed to the
+            // JSON unmarshalling method GetTokenStateAndContractId since the expiryTime will be elapsed for the
+            // claim to happen (i.e., if assetStatePointer is null, then expiryTimeSecs will be set to past time).
+            return ""
+        }
+
+        val assetStatePointer: StaticPointer<ContractState> = assetPledgeState.assetStatePointer!!
+        val assetState = assetStatePointer.resolve(serviceHub).state.data as FungibleToken
+
+        @Suppress("UNCHECKED_CAST")
+        val fungibleHouseTokenPointer: TokenPointer<FungibleHouseTokenState> = assetState.amount.token.tokenType as TokenPointer<FungibleHouseTokenState>
+        val fungibleHouseTokenState = fungibleHouseTokenPointer.pointer.resolve(serviceHub).state.data
+
+        println("Creating fungible token asset JSON from StatePointer.")
+        var marshalledAssetJson =
+            subFlow(MarshalFungibleToken(fungibleHouseTokenState.symbol, assetState.amount.quantity, assetPledgeState.lockerCert))
+
+        return marshalledAssetJson
+    }
+}
+
+/**
+ * The GetAssetPledgeStatusByPledgeId flow fetches the fungible tokens pledge status in the exporting network and returns as byte array.
+ * It is called during the interop query by importing network before performing the claim on fungible tokens pledged in exporting network.
+ *
+ * @property pledgeId The unique identifier representing the pledge on fungible tokens for transfer, in the exporting n/w.
+ * @property recipientNetworkId The id of the network in which the pledged fungible tokens will be claimed.
+ */
+@InitiatingFlow
+@StartableByRPC
+class GetAssetPledgeStatusByPledgeId(
+    val pledgeId: String,
+    val recipientNetworkId: String
+) : FlowLogic<ByteArray>() {
+    @Suspendable
+    override fun call(): ByteArray {
+
+        var assetPledgeState: AssetPledgeState = subFlow(GetAssetPledgeStatus(pledgeId, recipientNetworkId))
+        println("Obtained [AssetPledgeState] vault state: ${assetPledgeState}.\n")
+        val marshalledAssetJson = subFlow(GetHouseTokenJsonStringFromStatePointer(assetPledgeState))
```

Comment:
Even though on top you said its done, I still see subflows be used here. Did you not commit that?

Reply:
Ok, I have done earlier only for the SimpleAsset. Now did it for `house-token` corDapp too.